File: solutions/dart/protein-translation/1/lib/protein_translation.dart
class ProteinTranslation {
  List<String> translate(String str) {
    List<String> res = [];

    if (str == '') return res;
    for (var i = 0; i < str.length; i += 3) {
      var sub = str.substring(i, i + 3);
      switch (sub) {
        case 'UAA':
        case 'UAG':
        case 'UGA':
          return res;
        case 'AUG':
          res.add('Methionine');
          break;
        case 'UUU':
        case 'UUC':
          res.add('Phenylalanine');
          break;
        case 'UUA':
        case 'UUG':
          res.add('Leucine');
          break;
        case 'UCU':
        case 'UCC':
        case 'UCA':
        case 'UCG':
          res.add('Serine');
          break;
        case 'UAU':
        case 'UAC':
          res.add('Tyrosine');
          break;
        case 'UGU':
        case 'UGC':
          res.add('Cysteine');
          break;
        case 'UGG':
          res.add('Tryptophan');
          break;
        default:
          throw (ArgumentError());
      }
    }

    return res;
  }
}

File: solutions/python/protein-translation/1/protein_translation.py
def proteins(strand):
    res=[]
    for index in range(0, len(strand), 3):
        serie=strand[index:index+3]
        if serie=='AUG':
            res.append('Methionine')
        if serie in ['UUU', 'UUC']:
            res.append('Phenylalanine')
        if serie in ['UUA', 'UUG']:
            res.append('Leucine')
        if serie in ['UCU', 'UCC', 'UCA', 'UCG']:
            res.append('Serine')
        if serie in ['UAU', 'UAC']:
            res.append('Tyrosine')
        if serie in ['UGU', 'UGC']:
            res.append('Cysteine')
        if serie =='UGG':
            res.append('Tryptophan')
        if serie in ['UAA', 'UAG', 'UGA']:
            break


    return res